README.md

```diff
@@ -88,329 +91,366 @@ This opens the potential for [canonicalization attacks]. Parsers for certain for
+    Cid("bafkreidb2q3ktgtlm5yio7buj3sypyghjtfh5ernsteqmakf4p2c5bwmyi"),
+    Cid("bafkreic75ydg5vkw324oqkcmqltfvc3kivyngqkibjoysdwiilakh4z5fe"),
+    Cid("bafkreiffdiz6raf46zrr3b2usufgz5fo44aggmocz4zappr6khhhljcdpy"),
+  ]
+}).to_json()
 ```
 
 The signature is then checked against the above fields, which passes since there's only a `role: "user"` entry. The application then uses the original JSON with the `role: "admin"` entry.
 
-# 2 Safety
+# Safety
 
-Data that has already been parsed to an in-memory IPLD representation can be canonically encoded trivially: it has already been through a [parser / validator][Parse Don't Validate].
+Data already parsed to an in-memory representation can be canonically encoded trivially: it has already been through a [parser / validator][Parse Don't Validate].
 
-Data purporting to conform to an IPLD encoding (such as [DAG-JSON]) MUST be validated prior to signature verification. This MAY be as simple as round-trip decoding/encoding the JSON and checking that the hash matches. A validation error MUST be signalled if it does not match.
+Data purporting to conform to an IPLD encoding (such as [DAG-JSON]) MUST be validated prior to signature verification. This MAY be as simple as round-trip decoding/encoding the JSON and checking that the hash matches. A validation error MUST be signaled if it does not match.
 
-> [Implementers] may provide an opt-in for systems where round-trip determinism is a desireable [sic] feature and backward compatibility with old, non-strict data is unnecessary.
+> Implementers may provide an opt-in for systems where round-trip determinism is a desireable [sic] feature and backward compatibility with old, non-strict data is unnecessary.
 >
-> — [DAG-JSON Spec][DAG-JSON]
+> — [DAG-JSON Spec][DAG-JSON]
 
 As it is critical for guarding against various attacks, the assumptions around canonical encoding MUST be enforced.
 
-## 2.1 Signing CIDs
-
-Rather than validating the inline IPLD, replacing the data with a CID link to the content MAY be used instead. Note while this is very safe (as it is impractical to alter a signed hash), this approach mixes data layout with security, and may have a performance, disk, and networking impacts.
+# Format
 
-### 2.1.1 Caching & Invalidation
+A Varsig MUST have metadata about both the [signature] and [payload encoding] that was signed over. Either field MAY be composed of one or more segments. The number of segments MUST be determined by the first segment. Recursive sub-segments MAY be used.
 
-Signing CIDs has two additional caching consequences:
+Varsig itself MUST contain the following segments:
 
-1. Signing CIDs enables a simple strategy for caching validation by CID.
-2. Such a strategy also MAY require accounting for revocation of the signing keys themselves. In this case, the cache would need to include additional information about the signing key.
+* [Prefix]: The Varsig [multicodec] prefix `0x34`
+* [Version]: The Varsig version number `0x01`
```

Comment:
I'm happy to see the addition of a version number for the `varsig` encoding.  This may very well be the wrong spot for this question but ... In this section of the [UCAN spec](https://github.com/ucan-wg/spec#envelope), the `.1.h` field description says it's a `Varsig` v1 header.  In the example JSON just below the diagram, the value `NBIFEgEAcQ` decodes to a `Varsig` v0 that's missing its signature bytes.  The `go-varsig` implementation therefore currently allows `Varsig` v0 decoding that have the correct number of signature bytes *OR* with zero signature bytes (in which case both an error is returned as well as the decoded value - ugh).  I'm hoping you'll tell us to go fix our UCAN implementation and that we can deprecate the v0 code in `go-varsig`.

Reply:
Yes, feel free to deprecate any Varsig v0 code. I'll update the UCAN spec to fix that prefix, and I'll merge this PR momentarily to clear up the Varsig end 👍